docs/api.md

```diff
@@ -0,0 +1,131 @@
+# API Documentation
+
+## General
+
+### Errors
+
+All API errors will be in JSON format. An erroneous response will have the
+following shape:
+
+```txt
+400 Status Message
+-----
+{"name": "NameOfError", "message": "Human-readable message"}
+```
+
+### Providing an API version
+
+Any plain request to the API will tell you the following:
+
+```json
+{"name":"BadRequestError","message":"No valid API version provided."}
+```
+
+This is because every client has to provide the version of the API they wish to
+use. This allows the API to make backwards-incompatible changes to certain
+end-points, whilst maintaining backwards compatibility with older clients.
+
+The API Version can be provided in two ways. The recommended way is **to send
+an `Api-Version` header**. The alternative is to set the `_apiv` query
+parameter. The header or query parameter has to contain a fully qualified
+[semantic version](http://semver.org/), for example: `Api-Version: 1.1.0`. The
+version corresponds to the version in `package.json`.
+
+### Authentication
+
+#### `POST /auth`
+
+Request an authorization token pair:
+
+```txt
+POST /auth
+Api-Version: 0.1.0
+Content-Type: application/json
+-----
+{"username": "avaq", "password": "password123"}
+```
+
+```txt
+200 Ok
+Set-Cookie: token=<authorization_token>
+-----
+{"token": "<authorization_token>", "refresh": "<refresh_token>"}
```

Comment:
I think it's worth adding a more explicit note in the documentation that refresh tokens should be stored manually from the response, as no cookies are being set for it.